src/DotVVM.Framework/Hosting/Middlewares/DotvvmRoutingMiddleware.cs

```diff
@@ -61,14 +63,28 @@ private bool TryParseGooglebotHashbangEscapedFragment(string queryString, out st
             context.Route = route;
             context.Parameters = parameters;
 
+            var presenter = route.GetPresenter();
+            var filters = ActionFilterHelper.GetActionFilters<IRequestActionFilter>(presenter.GetType().GetTypeInfo());
+            filters.AddRange(context.Configuration.Runtime.GlobalFilters.OfType<IRequestActionFilter>());
             try
             {
-                await route.ProcessRequest(context);
+                foreach (var f in filters) await f.OnPageLoadingAsync(context);
+                await presenter.ProcessRequest(context);
+                foreach (var f in filters) await f.OnPageLoadedAsync(context);
             }
             catch (DotvvmInterruptRequestExecutionException)
             {
                 // the response has already been generated, do nothing
             }
+            catch(Exception exception)
+            {
+                foreach (var f in filters)
+                {
+                    await f.OnPageExceptionAsync(context, exception);
```

Comment:
DotvvmPresenter rethrows only `HttpException`, `DotvvmControlException` should be handled like any other. I'll fix it.